kernel/conversion.ml

```diff
@@ -477,6 +477,18 @@ and eqappr cv_pb l2r infos (lft1,st1) (lft2,st2) cuniv =
           else (* Two projections in WHNF: unfold *)
             raise NotConvertible)
 
+    | (FProj (p1,_r1,c1), FFlex(ConstKey (p2,_))) when Names.Constant.CanOrd.equal (Names.Projection.constant p1) p2 ->
```

Comment:
This is unsound when the constant is not generated as expected (which is done by the higher layers so the kernel must not rely on it).

Reply:
That is a very good point.